webapp/src/api.rs

```diff
@@ -188,37 +221,95 @@ pub async fn send_payment(
     Ok(())
 }
 
+#[utoipa::path(
+get,
+path = "/api/node",
+responses(
+(status = 200, description = "Get node id", body = String)
+)
+)]
 pub async fn get_node_id() -> impl IntoResponse {
     ln_dlc::get_node_pubkey().to_string()
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, ToSchema)]
 pub struct Seed {
     seed: Vec<String>,
 }
 
+#[utoipa::path(
+get,
+path = "/api/seed",
+responses(
+(status = 200, description = "Return seed phrase", body = Seed)
+)
+)]
 pub async fn get_seed_phrase() -> Json<Seed> {
     Json(Seed {
         seed: ln_dlc::get_seed_phrase(),
     })
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, ToSchema)]
 pub struct OrderId {
     id: Uuid,
 }
 
-#[derive(Deserialize, Clone)]
+#[derive(Serialize, Deserialize, ToSchema, Clone, Copy, Debug)]
+pub enum Direction {
+    Long,
+    Short,
+}
+
+impl From<trade::Direction> for Direction {
+    fn from(value: trade::Direction) -> Self {
+        match value {
+            trade::Direction::Long => Direction::Long,
+            trade::Direction::Short => Direction::Short,
+        }
+    }
+}
+
+impl From<Direction> for trade::Direction {
+    fn from(value: Direction) -> Self {
+        match value {
+            Direction::Long => trade::Direction::Long,
+            Direction::Short => trade::Direction::Short,
+        }
+    }
+}
+
+#[derive(Deserialize, Clone, ToSchema)]
 pub struct NewOrderParams {
     #[serde(with = "rust_decimal::serde::float")]
     pub leverage: Decimal,
     #[serde(with = "rust_decimal::serde::float")]
     pub quantity: Decimal,
     pub direction: Direction,
-    #[serde(with = "bitcoin::amount::serde::as_sat::opt")]
-    pub coordinator_reserve: Option<Amount>,
-    #[serde(with = "bitcoin::amount::serde::as_sat::opt")]
-    pub trader_reserve: Option<Amount>,
```

Comment:
😞 That's a bit annoying.

Reply:
I know :( We could potentially wrap it into another type and then imple `ToSchema` but I found this was the easiest way.